sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosException.java

```diff
@@ -551,6 +551,31 @@ void setRntbdPendingRequestQueueSize(int rntbdPendingRequestQueueSize) {
                 public CosmosException createCosmosException(int statusCode, Exception innerException) {
                     return new CosmosException(statusCode, innerException);
                 }
+
+                @Override
+                public CosmosException createSerializableCosmosException(CosmosException cosmosException) {
+                    if (cosmosException == null) {
+                        return null;
+                    }
+                    CosmosException exception = new CosmosException(cosmosException.statusCode,
+                        cosmosException.cosmosError, cosmosException.getResponseHeaders());
+                    exception.requestTimeline = cosmosException.requestTimeline;
+                    exception.channelAcquisitionTimeline = cosmosException.channelAcquisitionTimeline;
+                    exception.rntbdChannelTaskQueueSize = cosmosException.rntbdChannelTaskQueueSize;
+                    exception.rntbdEndpointStatistics = cosmosException.rntbdEndpointStatistics;
+                    exception.lsn = cosmosException.lsn;
+                    exception.partitionKeyRangeId = cosmosException.partitionKeyRangeId;
+                    exception.requestUri = cosmosException.requestUri;
+                    exception.resourceAddress = cosmosException.resourceAddress;
+                    exception.requestPayloadLength = cosmosException.requestPayloadLength;
+                    exception.rntbdPendingRequestQueueSize = cosmosException.rntbdPendingRequestQueueSize;
+                    exception.rntbdRequestLength = cosmosException.rntbdRequestLength;
+                    exception.rntbdResponseLength = cosmosException.rntbdResponseLength;
+                    exception.sendingRequestHasStarted = cosmosException.sendingRequestHasStarted;
+                    exception.requestHeaders = null;
```

Comment:
Why ignore the request headers? Espcially for 400 - Bad request I would expect request headers to be important? EVen for any 404/1002 read sessionnot available request headers are critical (and should be serialized in diagnostics)?

Reply:
Valid point, will add the request headers in next iteration.

Reply:
Discussed offline, we don't need requestHeaders in serializable CosmosException object.